### ISSUE TITLE
SVG Output

### ISSUE THREAD

**Issue by doctormo** [LemonPi/DAD#2]
This model creation is really interesting. I'd love to be able to export the results into svg for editing in Inkscape.

**Comment by LemonPi** [comment]
That's doable (by wrapping canvas's Context2D, which I already do to track pen movements and transforms), and there are libraries like [canvas2svg](https://github.com/gliffy/canvas2svg) that sort of does this. Although, that library does more than just wrap Context2D - it replaces canvas, so maybe [canvas-svg](https://code.google.com/archive/p/canvas-svg/)

This will be a lower priority task though, as I'm still finalizing the API of version 1.0 and documenting it. (As well as the serialization/deserialization to JSON).

Mind I ask what do you plan to do with the model after editing it in Inkscape?

**Comment by LemonPi** [comment]
OK I made a pull request to canvas2svg that should allow it to act more like a wrapper (instead of always creating its own canvas and ctx, if a Context2D is passed in, it'll just wrap around that instead)

**Comment by doctormo** [comment]
Sure, my plan was to have fun playing around with it for posing the different body shapes. Since it's html, this solution is much more likely to be svg accessible and thus editable.

One thing you might be able to do is if you're using svg instead of canvas for the rendering, you could add animation for free (no javascript required if you use SMIL :+1: )

**Comment by LemonPi** [comment]
I think it might be difficult to modify body shapes after exporting as external SVG due to the body's inherit symmetry. If you modify control points on one side of the body, you'd likely like to have it for the other side as well. Unfortunately Inkscape's current mirroring is not that great, and it'll be difficult to work with a full path that's already been mirrored (rather than a half path that you can then mirror).

I can see SVG export useful to embed static bodies in other people's work, but manipulating body shapes should best be done inside the tester/canvas playground. What do you find lacking about playing around with the body shapes inside the tester (that you'd need them to be in SVG format)? If by "posing the different body shapes" you mean actually putting them in different poses, then yeah I agree it's currently inadequate (only armRotation changes only pose; other are physical parameters).

**Comment by doctormo** [comment]
Yes, I mean to pose them like armRotation. But it's ok if it's not clean when I get it in there. Just a bit of play.
